src/auth.rs

```diff
@@ -87,9 +95,6 @@ pub fn authenticate_verify(
 	secret_key: &SecretKey,
 	data: &[u8],
 ) -> Result<bool, UnknownCryptoError> {
-	if secret_key.get_length() < BLAKE2B_MIN_KEY_SIZE {
-		return Err(UnknownCryptoError);
-	}
```

Comment:
This length check should not have been removed. We want to check in both functions.

Reply:
It seems I was looking at outdated changes here, but the other comments seem valid.